velox/exec/Window.cpp

```diff
@@ -41,16 +41,42 @@ void initKeyInfo(
   }
 }
 
-void checkDefaultWindowFrame(const core::WindowNode::Function& windowFunction) {
-  VELOX_CHECK_EQ(
-      windowFunction.frame.type, core::WindowNode::WindowType::kRange);
-  VELOX_CHECK_EQ(
-      windowFunction.frame.startType,
-      core::WindowNode::BoundType::kUnboundedPreceding);
-  VELOX_CHECK_EQ(
-      windowFunction.frame.endType, core::WindowNode::BoundType::kCurrentRow);
-  VELOX_CHECK_EQ(windowFunction.frame.startValue, nullptr);
-  VELOX_CHECK_EQ(windowFunction.frame.endValue, nullptr);
+// Update frameBound with the first row of the partition i.e zero for UNBOUNDED
+// PRECEDING frame bound.
+void findRowUnboundedPreceding(
+    vector_size_t* frameBound,
+    vector_size_t numRows) {
+  std::memset(frameBound, 0, numRows * sizeof(vector_size_t));
+}
+
+// Update frameBound with the first row of the peer group when frame start is
+// CURRENT ROW in RANGE mode.
+void findCurrentRowFrameStart(
+    vector_size_t* frameBound,
+    vector_size_t partitionStartRow,
+    vector_size_t* peerStarts,
+    vector_size_t numRows) {
+  std::copy(peerStarts, peerStarts + numRows, frameBound);
+}
+
+// Update frameBound with the last row of the peer group when frame end is
+// CURRENT ROW in RANGE mode.
+void findCurrentRowFrameEnd(
+    vector_size_t* frameBound,
+    vector_size_t partitionStartRow,
+    vector_size_t* peerEnds,
+    vector_size_t numRows) {
+  std::copy(peerEnds, peerEnds + numRows, frameBound);
+}
+
+// Update frameBound with the last row of the partition for UNBOUNDED FOLLOWING
+// frame bound.
+void findRowUnboundedFollowing(
+    vector_size_t* frameBound,
+    vector_size_t partitionStartRow,
```

Comment:
You can simply send a lastRowIndex parameter which is assigned to (partitionEndRow - partitionStartRow) here.

Reply:
Thanks for the suggestion, updated.